components/server/src/test/java/com/hotels/styx/server/netty/codec/NettyToStyxRequestDecoderTest.java

```diff
@@ -222,17 +221,23 @@ public void throwsBadRequestExceptionWhenRequestMessageContainsMoreThanOneHostHe
 
     @Test
     public void callsTheEscaperForUnwiseChars() throws Exception {
-        UnwiseCharsEncoder encoder = mock(UnwiseCharsEncoder.class);
+        AtomicBoolean encoderCalled = new AtomicBoolean();
+        UnwiseCharsEncoder encoder = x -> {
```

Comment:
Why replace mocking with this approach?

Reply:
I meant to comment on the same to be honest... I think keeping mock.verify() would make our code a bit more uniform.

Reply:
By mistake. I have backed off this change.